src/Microsoft.Identity.Client/Internal/Requests/BaseRequest.cs

```diff
@@ -118,6 +120,10 @@ protected void ValidateScopeInput(SortedSet<string> scopesToValidate)
                 RequestContext.Logger.Error(ex);
```

Comment:
This particular `catch (Exception ex)` code branch is pre-existing code, which I did not touch. I think it does nothing more than trying to log.

What I changed is in the green lines. Basically create a request id, and later flush it in the `finally` clause. And, this `RunAsync()` will always be run, because none of the BaseRequest sub-classes overrides this method (and they shouldn't/wouldn't).

(PS: I'm currently in head-down mode hacking the upcoming ApiEvent PR, so this response is short. Feel free to let me know if you would like an in-person conversation for this one.)